java/build-android.gradle

```diff
@@ -160,7 +168,22 @@ publishing {
 						id = 'onnxruntime'
 						name = 'ONNX Runtime'
 						email = 'onnxruntime@microsoft.com'
-					}
+                    }
+                }
+
+            if (addQnnDependency) {
+				println "Modifying the POM XML to include QNN dependency"
+				def qnnVersion = System.properties['qnnVersion']
+
+				qnnVersion = "$qnnVersion.+"
```

Comment:
Can we limit this to `major.minor.*` to be safe? Might need to process the input qnnVersion to truncate at the second '.' so the `.patch[.build_id]` part, if provided, is ignored.

Could we also read qnnVersion up the top where we process all the other System.properties values so that processing is all in one place? That would also be a good place to add a comment explaining where the version number comes from and the expected formatting.

I assume if the value is not set we just end up with an empty qnnVersion. if so, we don't need to check addQnnDependency before reading the value.

Reply:
That's a good point. Updated the logic to read the version for sdk.yaml, truncate and add to the depedency. The [pom file](https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=570063&view=artifacts&pathAsName=false&type=publishedArtifacts) now has 2.26.*

Reply:
Brett from Qualcomm team recommended using the exact version of patch too. That would mean downgrading the SDK version we have right now- does that work? I tested with 2.24 SDK too, there are no failing tests

Reply:
Updated the code to check for major.minor.patch according to Qualcomm's recommendation